### PR TITLE
Serilog dupe stackframe fix

### LINKED ISSUE #1070: Getting duplicate logs using log forwarding and Serilog
**Description**
Seeing duplicate log entries when using Serilog and log forwarding.

**Expected Behavior**
Logs should be 1:1 not 1:2.

**[Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/diagnostics-cli-nrdiag/diagnostics-cli-nrdiag/) results**

- Disabling forwarding results in no messages
- Disabling infra still results in 2 messages

**Steps to Reproduce**

**Your Environment**
```
App: .NET Fr. 4.6.2
    Agent 9.7.1
    Infra with event logs collection
    Serilog 2.0.0
        Enrichers.
            Environment
            process
            Thread
        Sinks
            File
            PeriodicBatching
            RollingFile
            Seq
    SerilogWeb.Classic 5.0.0.0
        .Mvc 2.0
        .WebApi **4.0**
```

### PR DESCRIPTION
**Pull request by jaffinito**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

Updates serilog wrapper to check for duplicate log entries by looking at the stackframes and checking for additional "Dispatch" methods.

There are two different versions of the stack trace, one that places the initial Dispatch method at 7 and one that places it at 8.  The logic in the wrapper handles this.

Tests all pass and performance is acceptable (let me know if you want to go over it)

Closes #1070 

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [x] Performance testing completed with satisfactory results (if required)
- [x] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [x] Perform code review
- [x] Pull request was adequately tested (new/existing tests, performance tests)
- [x] Review Changelog

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Logging/SerilogWrapper.cs [hunk 1]
```diff
@@ -37,6 +38,26 @@ public CanWrapResponse CanWrap(InstrumentedMethodInfo methodInfo)
 
         public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall instrumentedMethodCall, IAgent agent, ITransaction transaction)
         {
+            var frame7 = new System.Diagnostics.StackFrame(7, false).GetMethod().Name;
```
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Logging/SerilogWrapper.cs:41]: I think a comment here explaining the issue would be great

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
I think it would be vital to cover both of these stack depth use cases in integration tests so we don't introduce a regression or new bug if we refactor the current approach.

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Logging/SerilogWrapper.cs (hunk 1 above)
Comment (nr-ahemsath) [reply src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Logging/SerilogWrapper.cs:41]: Yes, a comment would be helpful.

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
Thank you for adding comments! There are a few minor typos in them but I can live with that.

This needs integration test coverage still.

**Review by vuqtran88 — APPROVED** [review]
We did a code walk through and addressed all of the concerned. This is good to go.

**Review by JcolemanNR — APPROVED** [review]
Great fix! Reviewed as a team and I approve.

**Pull request merged**